contracts/contracts/StealthKeyRegistry.sol

```diff
@@ -0,0 +1,201 @@
+   * chainId and the contract address
+   */
+  constructor() {
+    uint256 _chainId;
+    assembly {
+      _chainId := chainid()
+    }
+
+    DOMAIN_SEPARATOR = keccak256(
+      abi.encode(
+        EIP712DOMAIN_TYPEHASH,
+        keccak256(bytes("Umbra Stealth Key Registry")),
+        keccak256(bytes("1")),
+        _chainId,
+        address(this)
+      )
+    );
+  }
+
+  // ======================================= Set Keys ===============================================
+
+  /**
+   * Sets the stealth keys associated with an address, for anonymous sends.
+   * May only be called by the associated address.
+   * @param _spendingPubKeyPrefix Prefix of the spending public key (2 or 3)
+   * @param _spendingPubKey The public key for generating a stealth address
+   * @param _viewingPubKeyPrefix Prefix of the viewing public key (2 or 3)
+   * @param _viewingPubKey The public key to use for encryption
+   */
+  function setStealthKeys(
+    uint256 _spendingPubKeyPrefix,
+    uint256 _spendingPubKey ,
+    uint256 _viewingPubKeyPrefix,
+    uint256 _viewingPubKey
+  ) external {
+    _setStealthKeys(msg.sender, _spendingPubKeyPrefix, _spendingPubKey , _viewingPubKeyPrefix, _viewingPubKey);
+  }
+
+  /**
+   * Sets the stealth keys associated with an address, for anonymous sends.
+   * May only be called by the associated address.
+   * @param _registrant The address for which stealth keys are being registered, must also sign
+   * @param _spendingPubKeyPrefix Prefix of the spending public key (2 or 3)
+   * @param _spendingPubKey The public key for generating a stealth address
+   * @param _viewingPubKeyPrefix Prefix of the viewing public key (2 or 3)
+   * @param _viewingPubKey The public key to use for encryption
+   * @param _v ECDSA signature component: Parity of the `y` coordinate of point `R`
+   * @param _r ECDSA signature component: x-coordinate of `R`
+   * @param _s ECDSA signature component: `s` value of the signature
+   */
+  function setStealthKeysOnBehalf(
+    address _registrant,
+    uint256 _spendingPubKeyPrefix,
+    uint256 _spendingPubKey ,
+    uint256 _viewingPubKeyPrefix,
+    uint256 _viewingPubKey,
+    uint8 _v,
+    bytes32 _r,
+    bytes32 _s
+  ) external {
+    // create EIP-712 Digest
+    bytes32 _digest =
+      keccak256(
+        abi.encodePacked(
+          "\x19\x01",
+          DOMAIN_SEPARATOR,
+          keccak256(
+            abi.encode(STEALTHKEYS_TYPEHASH, _spendingPubKeyPrefix, _spendingPubKey , _viewingPubKeyPrefix, _viewingPubKey)
+          )
+        )
+      );
+
+    // recover the signing address and ensure it matches the registrant
+    address _recovered = ecrecover(_digest, _v, _r, _s);
+    require(_recovered == _registrant, "StealthKeyRegistry: Invalid Signature");
+
+    // now that we know the registrant has authorized it, update the stealth keys
+    _setStealthKeys(_registrant, _spendingPubKeyPrefix, _spendingPubKey , _viewingPubKeyPrefix, _viewingPubKey);
```

Comment:
Why do your `prettier` commit hooks never run haha. Can you run `yarn prettier` before merging? Seems there's other formatting that it changes also
```suggestion
    _setStealthKeys(_registrant, _spendingPubKeyPrefix, _spendingPubKey, _viewingPubKeyPrefix, _viewingPubKey);
```